src/test/java/org/opentripplanner/graph_builder/module/osm/WalkableAreaBuilderTest.java

```diff
@@ -25,26 +27,24 @@
 import org.opentripplanner.street.model.edge.AreaEdge;
 import org.opentripplanner.street.model.vertex.VertexLabel;
 import org.opentripplanner.street.model.vertex.VertexLabel.OsmNodeOnLevelLabel;
-import org.opentripplanner.transit.model.framework.Deduplicator;
 
+@Execution(ExecutionMode.SAME_THREAD)
```

Comment:
Is this or other changes that were made in order to get tests to work before the final change still required for the tests to work?

Reply:
Seems to work without it as well.

Reply:
This line is not necessary but the rest of the changes is IMO worthwhile even if it doesn't fix the non-determinism, which I think was caused somehow by maxAreaNodes and the visibility vertices.